### PR TITLE
[runtime] Fix the mapping of 'int' in jit icall signatures, it should be int32 not native int.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by jaykrell** [comment]
Why? What is the difference?
Just hypothetical for systems with 16bit or 64bit int, or something else?

**Comment by jaykrell** [comment]
Oh, in this context, I guess the names are confusing, and "int" meant "size_t".

File: mono/metadata/icall.c [hunk 1]
```diff
@@ -9517,12 +9517,12 @@ typedef enum ICallSigType {
 	ICALL_SIG_TYPE_float    = 0x02,
 	ICALL_SIG_TYPE_int      = 0x03,
 	ICALL_SIG_TYPE_int16    = 0x04,
-	ICALL_SIG_TYPE_int32    = 0x05,
-	ICALL_SIG_TYPE_int8     = 0x06,
-	ICALL_SIG_TYPE_long     = 0x07,
+	ICALL_SIG_TYPE_int32    = ICALL_SIG_TYPE_int,
+	ICALL_SIG_TYPE_int8     = 0x05,
+	ICALL_SIG_TYPE_long     = 0x06,
+	ICALL_SIG_TYPE_ptr      = 0x07,
```
Comment (jaykrell) [inline mono/metadata/icall.c]: Alpaphetical sort got broken.

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
The windows failures look relevant. Some GC icalls are defined as:
	mono_register_jit_icall (mono_gc_alloc_obj, mono_icall_sig_object_ptr_int, FALSE);
	mono_register_jit_icall (mono_gc_alloc_vector, mono_icall_sig_object_ptr_int_int, FALSE);
	mono_register_jit_icall (mono_gc_alloc_string, mono_icall_sig_object_ptr_int_int32, FALSE);
	mono_register_jit_icall (mono_profiler_raise_gc_allocation, mono_icall_sig_void_object, FALSE);

but they take size_t arguments, not int/pointer.

**Comment by jaykrell** [comment]
Right, I noticed this a while ago.
“int” in this context means “ptrdiff_t”.
I preserved that when I “rewrote” it.

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
Reviewed the other signatures and they looked ok. 'int' was always used to int32 in almost all cases.

**Pull request merged**